pkg/controller/discovery/model/cluster.go

```
@@ -278,6 +278,7 @@ func (m *Cluster) NsList(db DB, page *Page) ([]*Namespace, error) {
 			&ns.Cluster)
 		if err != nil {
 			Log.Trace(err)
+			return nil, err
```

Should return the error here.

(same below)